Adding SCVMM API Version 2024-06-01

kartikverma020 opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

kartikverma020 committed:
Base commit. Adding copy of the existing api version

kartikverma020 committed:
Updating readme.md

kartikverma020 committed:
Updating main.tsp and models.tsp

kartikverma020 committed:
Updating scvmm.json

kartikverma020 committed:
Updating examples for api version 2024-06-01

pratikjoshi14 commented on specification/scvmm/ScVmm.Management/models.tsp:
```diff
@@ -640,6 +643,36 @@ model OsProfileForVmInstance {
+
+  /** Gets or sets the index value of the timezone. */
+  @added(Versions.v2024_06_01)
+  timezone?: int32;
```
timezone is generally a string in other APIs.. did you mean timezoneOffset here?

kartikverma020 replied:
It is expected that the entries in timezone will be the index values in https://learn.microsoft.com/en-us/previous-versions/windows/embedded/ms912391(v=winembedded.11).

venu-sivanadham replied:
though the comment says its time zone index, but the property name itself is misleading. 
Do you want to rename the property? like timezoneIndex

pratikjoshi14 commented on specification/scvmm/ScVmm.Management/models.tsp:
```diff
@@ -640,6 +643,36 @@ model OsProfileForVmInstance {
+
+  /** Gets or sets the product key.Input format xxxxx-xxxxx-xxxxx-xxxxx-xxxxx */
+  @added(Versions.v2024_06_01)
+  productKey?: string;
```
Any @pattern here?

kartikverma020 replied:
There is an expected format for the data received from portal however there is no such check as the command itself will throw the error of invalid product key.

pratikjoshi14 commented on specification/scvmm/resource-manager/Microsoft.ScVmm/stable/2024-06-01/scvmm.json:
```diff
@@ -4174,6 +4186,41 @@
+          "format": "int32",
+          "description": "Gets or sets the index value of the timezone."
+        },
+        "runOnceCommands": {
```
If these commands are run only during creation, the values will be persisted inside object forever? What would be the behavior during update operation?

pratikjoshi14 commented on specification/scvmm/ScVmm.Management/models.tsp:
```diff
@@ -640,6 +643,36 @@ model OsProfileForVmInstance {
+
+  /** Get or sets the commands to be run once at the time of creation separated by semicolons. */
+  @added(Versions.v2024_06_01)
+  runOnceCommands?: string;
```
If these commands are run only during creation, the values will be persisted inside object forever? What would be the behavior during update operation?

kartikverma020 replied:
Answer to your first question Yes during get operations we will be able to the see the commands.
Second question - These added properties are inside the Osprofile which is only present in VirtualMachineInstanceProperties and not in the VirtualMachineInstanceUpdateProperties.Thus the user wont be able to change any properties under the os profile.

venu-sivanadham commented on specification/scvmm/ScVmm.Management/examples/2024-06-01/GuestAgents_Get_MaximumSet_Gen.json:
```diff
@@ -0,0 +1,38 @@
+          "provisioningAction": "install",
+          "status": "jpoukrzfenzrmjdahimkl",
+          "customResourceName": "mhqymxkapuvsugd",
+          "provisioningState": "Succeeded"
```
privateLinkScopeResourceId property missing

kartikverma020 replied:
Done.

venu-sivanadham commented on specification/scvmm/ScVmm.Management/examples/2024-06-01/GuestAgents_ListByVirtualMachineInstance_MaximumSet_Gen.json:
```diff
@@ -0,0 +1,43 @@
+              "provisioningAction": "install",
+              "status": "jpoukrzfenzrmjdahimkl",
+              "customResourceName": "mhqymxkapuvsugd",
+              "provisioningState": "Succeeded"
```
nit: you can add "privateLinkScopeResourceId" here as well.

kartikverma020 replied:
Done

venu-sivanadham commented on specification/scvmm/ScVmm.Management/models.tsp:
```diff
@@ -640,6 +643,36 @@ model OsProfileForVmInstance {
+
+  /** Gets or sets the product key.Input format xxxxx-xxxxx-xxxxx-xxxxx-xxxxx */
+  @added(Versions.v2024_06_01)
+  productKey?: string;
```
>productKey [](http://example.com/codeflow?start=2&length=10)

does it quality for a secret? you want to return the product-key it in get/read calls?

kartikverma020 replied:
Have made it a secret.

venu-sivanadham commented on specification/scvmm/ScVmm.Management/examples/2024-06-01/VirtualMachineInstances_CreateOrUpdate_MaximumSet_Gen.json:
```diff
@@ -0,0 +1,329 @@
+            "computerName": "uuxpcxuxcufllc",
+            "osType": "Windows",
+            "osSku": "cxqnjxgkts",
+            "osVersion": "djt"
```
newly added osProfile properties are missing in this response.

venu-sivanadham replied:
Update Get & List APIs as well with newly added properties.

kartikverma020 replied:
Done

venu-sivanadham commented:
Update your PR, its out-of-date with base branch.

kartikverma020 committed:
Resolving Comments.

Adding privatelinkscoperesourceid in get and list and update osprofilepropeties in virtualmachine instance get and list.

kartikverma020 committed:
Merge branch 'main' into vermakartik/2024-06-01

kartikverma020 committed:
Making productKey a secret.

kartikverma020 committed:
Merge branch 'main' into vermakartik/2024-06-01

venu-sivanadham reviewed (approved):
:shipit:

kartikverma020 commented:
/pr RequestMerge
